sound/soc/sof/debug-dsp-ops.c

```diff
@@ -8,6 +8,61 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+/*
+ * set dsp power state op by writing the requested power state.
+ * ex: echo D3 > dsp_power_state
+ */
+static int sof_dsp_ops_set_power_state(struct snd_sof_dev *sdev, char *state)
+{
+	/* only D0 and D3 supported for now */
+	if (strcmp(state, "D3") && strcmp(state, "D0")) {
+		dev_err(sdev->dev, "Unsupported state %s\n", state);
+		return -EINVAL;
+	}
+
+	if (!strcmp(state, "D0")) {
+		/* power on the DSP if it is in D3 */
+		if (sdev->dsp_power_state.state == SOF_DSP_PM_D3) {
+			struct sof_dsp_power_state target_state = {
+				.state = SOF_DSP_PM_D0,
+			};
+
+			/*
+			 * Fix set_power_state to power up the DSP.
+			 * ATM this is taken care of during firmware boot
```

Comment:
I still don't get the meaning of this comment....

Reply:
today we handle powering up the DSP during snd_sof_run_firmware(). So, if you want to go from D3 -> D0, we have to write to boot_fw() and it will handle it automatically. I want to fix this later on to split the DSP powering up out of the run_firmware call. But this is not critical right now

Reply:
but the power up is a bit more than just toggle a switch, we need to 'send IPC' (the boot config needs to be placed), we need to wait for IPC (response that the DSP is ready to either start the download or it booted with IMR.

Is it worth adding confusing comments for something that will happen at some point. Or not.

Reply:
I was also unable to understand the comment.